lib/ruby/stdlib/date/format.rb

```diff
@@ -376,9 +376,7 @@ def self._strptime_i(str, fmt, e) # :nodoc:
   private_class_method :_strptime_i
 
   def self._strptime(str, fmt='%F')
-    parser = org.jruby.util.RubyDateParser.new
-    map = parser.parse(JRuby.runtime.current_context, fmt, str)
-    return map.nil? ? nil : map.to_hash.inject({}){|hash,(k,v)| hash[k.to_sym] = v; hash}
+    return JRuby.runtime.current_context.getRubyDateParser.parseAndCreateHash(JRuby.runtime.current_context, fmt, str)
```

Comment:
@muga I think you could make parser.parse just return the symboled hash right?  Then @kares comment about currentContext (and having to know there is a cache) can be encapsulated by the parse.

As a broader comment.  A Bag is made then a Map is made and then a Hash with Symbols is made.  I have not viewed how much a Bag modifies its data but I wondered if you could just drop the bag and the map and just use a Hash with symbols directly as you parse?  I can see some post processing happening at end of bag and when map is made but I just wondered about whether this could just be one thing or not.